packages/react-tinacms-github/src/open-authoring/OpenAuthoringProvider.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import React, { useState, useEffect } from 'react'
+import { getForkName, getHeadBranch } from './repository'
+import { useCMS } from 'tinacms'
+import OpenAuthoringErrorModal from '../github-error/OpenAuthoringErrorModal'
+import OpenAuthoringAuthModal from './OpenAuthoringAuthModal'
+import { OpenAuthoringContext } from './OpenAuthoringContext'
+import { useOpenAuthoring } from './useOpenAuthoring'
+
+interface ProviderProps {
+  children: any
+  authenticate: () => Promise<void>
+  enterEditMode: () => void
+  exitEditMode: () => void
+  error?: any
+}
+
+interface AuthState {
+  authenticated: true
+  forkValid: true
+}
+
+export const OpenAuthoringProvider = ({
```

Comment:
The name `OpenAuthoringProvider` seems both too broad and too specific to me. 

Too broad because it suggests it's a general solution to Open Authoring,  which it is not too.

Too specific because if this supports private repos (i.e. no forking, only authorized if you have access to the repo) in the future then it wouldn't be just for open authoring. 

What do you think of renaming to something like `GithubAuthenticationProvider`?

Reply:
Since it also handles entering/exiting editing mode maybe `TinacmsGithubProvider`?